docbrown/core/src/tgraph.rs

```diff
@@ -357,102 +465,337 @@ impl TemporalGraph {
+                    dst_g_id: v,
+                    src_id: dst,
                     dst_id: v_pid,
                     t: None,
-                    g: self,
                     e_meta,
                 },
             )),
             Direction::BOTH => Box::new(itertools::chain!(
-                self.neighbours_window(v, w, Direction::IN),
-                self.neighbours_window(v, w, Direction::OUT)
+                self.edges_window(v, w, Direction::IN),
+                self.edges_window(v, w, Direction::OUT)
             )),
         }
     }
 
-    pub(crate) fn neighbours_window_t(
+    pub(crate) fn edges_window_t(
         &self,
         v: u64,
-        r: &Range<i64>,
+        w: &Range<i64>,
         d: Direction,
-    ) -> Box<dyn Iterator<Item = EdgeView<'_, Self>> + Send + '_> {
+    ) -> Box<dyn Iterator<Item = EdgeView> + Send + '_> {
         let v_pid = self.logical_to_physical[&v];
 
         match d {
-            Direction::OUT => Box::new(self.neighbours_iter_window_t(v_pid, r, d).map(
-                move |(v, t, e_meta)| EdgeView {
+            Direction::OUT => Box::new(self.edges_iter_window_t(v_pid, w, d).map(
+                move |(dst, t, e_meta)| EdgeView {
+                    src_g_id: v,
+                    dst_g_id: self.v_g_id(dst, e_meta),
                     src_id: v_pid,
-                    dst_id: v,
+                    dst_id: dst,
                     t: Some(t),
-                    g: self,
                     e_meta,
                 },
             )),
-            Direction::IN => Box::new(self.neighbours_iter_window_t(v_pid, r, d).map(
-                move |(v, t, e_meta)| EdgeView {
-                    src_id: v,
+            Direction::IN => Box::new(self.edges_iter_window_t(v_pid, w, d).map(
+                move |(dst, t, e_meta)| EdgeView {
+                    src_g_id: self.v_g_id(dst, e_meta),
+                    dst_g_id: v,
+                    src_id: dst,
                     dst_id: v_pid,
                     t: Some(t),
-                    g: self,
                     e_meta,
                 },
             )),
             Direction::BOTH => Box::new(itertools::chain!(
-                self.neighbours_window_t(v, r, Direction::IN),
-                self.neighbours_window_t(v, r, Direction::OUT)
+                self.edges_window_t(v, w, Direction::IN),
+                self.edges_window_t(v, w, Direction::OUT)
             )),
         }
     }
+
+    pub(crate) fn neighbours(
+        &self,
+        v: u64,
+        d: Direction,
+    ) -> Box<dyn Iterator<Item = VertexView> + Send + '_>
+    where
+        Self: Sized,
+    {
+        let edges = self.edges(v, d);
+
+        Box::new(edges.map(move |edge| {
+            let EdgeView { src_id, dst_id, .. } = edge;
+
+            let src_g_id = edge.src_g_id;
+            let dst_g_id = edge.dst_g_id;
+
+            if v == src_g_id {
+                VertexView::new(dst_g_id, dst_id)
```

Comment:
This fine even if the vertex isn't on the same shard?

Reply:
I'll write a test for this!